packages/typespec-ts/test/modularIntegration/arrayItemTypes.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { assert } from "chai";
+import { ArrayClient } from "./generated/arrays/items/src/index.js";
+
+interface TypeDetail {
+  type: string;
+  defaultValue: any;
+  convertedToFn?: (_: any) => any;
+}
+
+const testedTypes: TypeDetail[] = [
+  {
+    type: "int32",
+    defaultValue: [1, 2]
+  },
+  {
+    type: "int64",
+    defaultValue: [Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER]
+  },
+  {
+    type: "boolean",
+    defaultValue: [true, false]
+  },
+  {
+    type: "string",
+    defaultValue: ["hello", ""]
+  },
+  {
+    type: "float32",
+    defaultValue: [42.42]
+  },
+  {
+    type: "datetime",
+    defaultValue: [new Date("2022-08-26T18:38:00Z")]
+  },
+  {
+    type: "duration",
+    defaultValue: ["P123DT22H14M12.011S"]
+  },
+  {
+    type: "unknown",
+    defaultValue: [1, "hello", null]
+  },
+  {
+    type: "model",
+    defaultValue: [
+      { property: "hello", children: undefined },
+      { property: "world", children: undefined }
```

Comment:
the children: undefined are added by the current serialization and deserialization logic in modular.